src/main/java/net/neoforged/neoforge/registries/datamaps/builtin/FurnaceFuel.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.registries.datamaps.builtin;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.util.ExtraCodecs;
+
+/**
+ * Data map value for {@linkplain NeoForgeDataMaps#FURNACE_FUELS furnace fuels}.
+ *
+ * @param burnTime how long (in ticks) the item will burn for
+ */
+public record FurnaceFuel(int burnTime) {
+    public static final Codec<FurnaceFuel> BURN_TIME_CODEC = ExtraCodecs.POSITIVE_INT
```

Comment:
shouldn't the burn time be limited to `Short.MAX_VALUE`? unless we patched furnaces to support integers yet.

Reply:
[We have indeed patched it.](https://github.com/neoforged/NeoForge/blob/1.20.x/patches%2Fnet%2Fminecraft%2Fworld%2Flevel%2Fblock%2Fentity%2FAbstractFurnaceBlockEntity.java.patch#L42-L44)

Reply:
Have we patched the sync too? DataSlots actually use shorts too.

Reply:
![image](https://github.com/neoforged/NeoForge/assets/13494793/46d7cb50-f8dc-4f91-9d90-787daaf44d05)
This needs to be patched so it sends smaller values but the client still sees the correct state.

Reply:
The client is never really going to see correct information if the values have to be truncated.  The only real solution would be to change `ClientboundContainerSetDataPacket` to use integers instead of shorts, but that will break protocol.  We could _potentially_ send a packet that encodes/decodes as int if we know the connection is NF<->NF

Reply:
The client only cares about litTime/litDuration. It doesn't really matter which values it receives as long as the ratio is correct.